extensions/amp-base-carousel/0.1/test/test-carousel.js

```diff
@@ -298,23 +308,23 @@ describes.realWin('carousel implementation', {}, (env) => {
         slideCount: 3,
         initialIndex: -1,
       });
-      expect(carousel.isAtStart()).to.be.true;
+      expect(carousel.scrollContainer_.scrollLeft).to.equal(0);
     });
 
     it('should start at slide 0 with initialIndex that is greater than number of slides', async () => {
       const carousel = await createCarousel({
         slideCount: 3,
         initialIndex: 4,
       });
-      expect(carousel.isAtStart()).to.be.true;
+      expect(carousel.scrollContainer_.scrollLeft).to.equal(0);
```

Comment:
@caroqliu So actually, using initial index 4 breaks amp-base-carousel: [repro](https://playground.amp.dev/?mode=Responsive#share=PCFET0NUWVBFIGh0bWw+CjxodG1sIOKaoT4KICA8aGVhZD4KICAgIDxtZXRhIGNoYXJzZXQ9InV0Zi04IiAvPgogICAgPHRpdGxlPkFNUCBMaWdodGJveCBDYXJvdXNlbCBXaXRoIEFNUCBTZWxlY3RvcjwvdGl0bGU+CiAgICA8bGluayByZWw9ImNhbm9uaWNhbCIgaHJlZj0ic2VsZi5odG1sIiAvPgogICAgPG1ldGEgbmFtZT0idmlld3BvcnQiIGNvbnRlbnQ9IndpZHRoPWRldmljZS13aWR0aCIgLz4KICAgIDxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoOy1tb3otYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7LW1zLWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoO2FuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RofUAtd2Via2l0LWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW1vei1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1tcy1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1vLWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1Aa2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fTwvc3R5bGU+PG5vc2NyaXB0PjxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjpub25lOy1tb3otYW5pbWF0aW9uOm5vbmU7LW1zLWFuaW1hdGlvbjpub25lO2FuaW1hdGlvbjpub25lfTwvc3R5bGU+PC9ub3NjcmlwdD4KICAgIDxzY3JpcHQgYXN5bmMgc3JjPSJodHRwczovL2Nkbi5hbXBwcm9qZWN0Lm9yZy92MC5qcyI+PC9zY3JpcHQ+CiAgICA8c2NyaXB0IGFzeW5jIGN1c3RvbS1lbGVtZW50PSJhbXAtY2Fyb3VzZWwiIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAvYW1wLWNhcm91c2VsLTAuMi5qcyI+PC9zY3JpcHQ+CiAgICA8c3R5bGUgYW1wLWN1c3RvbT4KICAgIAogICAgPC9zdHlsZT4KICA8L2hlYWQ+CiAgPGJvZHk+CiAgPGFtcC1jYXJvdXNlbCBpZD0iY2Fyb3VzZWwxIiB3aWR0aD0iNDUwIiBoZWlnaHQ9IjMwMCIgdHlwZT0ic2xpZGVzIiBzbGlkZT0iNCI+CiAgICAgICAgICA8YW1wLWltZyBzcmM9Ii9zdGF0aWMvaW5saW5lLWV4YW1wbGVzL2ltYWdlcy9pbWFnZTEuanBnIgogICAgd2lkdGg9IjQ1MCIKICAgIGhlaWdodD0iMzAwIj48L2FtcC1pbWc+CiAgPGFtcC1pbWcgc3JjPSIvc3RhdGljL2lubGluZS1leGFtcGxlcy9pbWFnZXMvaW1hZ2UyLmpwZyIKICAgIHdpZHRoPSI0NTAiCiAgICBoZWlnaHQ9IjMwMCI+PC9hbXAtaW1nPgogIDxhbXAtaW1nIHNyYz0iL3N0YXRpYy9pbmxpbmUtZXhhbXBsZXMvaW1hZ2VzL2ltYWdlMy5qcGciCiAgICB3aWR0aD0iNDUwIgogICAgaGVpZ2h0PSIzMDAiPjwvYW1wLWltZz4KICA8L2FtcC1jYXJvdXNlbD4KPC9ib2R5Pgo8L2h0bWw+Cg==)

I've filed https://github.com/ampproject/amphtml/issues/33512 to tackle this problem in amp-carousel 0.2. 

IMO we either remove this test or fix it in this PR or fix it in a later PR.

Reply:
Wanted to unblock you here, fixed in #33512.